### ISSUE TITLE
Nested subcommand support with `--help`

### ISSUE THREAD

**Issue by nathan-at-least** [murarth/gumdrop#29]
Using `gumpdrop 0.7.0` with `#[derive(Options)]` does not support nested subcommands in the built-in help output.

For an example program, here's `myprogram` which allows users to manage user accounts and organizations with a command structure similar to this:

```shell
$ mycommand account new
$ mycommand account update --name alice
$ mycommand org add-member
```

etc...

The problem is that the behavior of `--help` seems unaware of subcommands. Here's an example main demonstrating the behavior for the above command:

```rust
use gumdrop::Options;

// Define options for the program.
#[derive(Debug, Options)]
pub struct MainOptions {
    #[options(help = "print help message")]
    help: bool,

    #[options(command)]
    command: Option<Command>,
}

#[derive(Debug, Options)]
pub enum Command {
    #[options(help = "manage an account")]
    Account(AccountOptions),

    #[options(help = "manage an organization")]
    Org(OrgOptions),
}

#[derive(Debug, Options)]
pub struct AccountOptions {
    #[options(help = "print help message")]
    help: bool,

    #[options(command)]
    command: Option<AccountCommand>,
}

#[derive(Debug, Options)]
pub enum AccountCommand {
    #[options(help = "create a new account")]
    New(AccountNewOptions),

    #[options(help = "update account details")]
    Update(AccountUpdateOptions),
}

#[derive(Debug, Options)]
pub struct AccountNewOptions {
    #[options(help = "print help message")]
    help: bool,
}

#[derive(Debug, Options)]
pub struct AccountUpdateOptions {
    #[options(help = "print help message")]
    help: bool,

    #[options(help = "set account name")]
    name: String,
}

// This is just empty scaffolding for now:
#[derive(Debug, Options)]
pub struct OrgOptions {}

fn main() {
    let opts = MainOptions::parse_args_default_or_exit();

    println!("{:#?}", opts);
}
```

The following snippets show the behavior of `--help` along with desired changes.

```
$ mycommand --help
Usage: mycommand [OPTIONS]

Optional arguments:
  -h, --help  print help message

Available commands:

  account  manage an account
  org      manage an organization
```

This output works well and supports the top-level subcommands well.
```
$ mycommand --help account
Usage: mycommand account [OPTIONS]

Optional arguments:
  -h, --help  print help message
```

This output shows the right commandline to include the `account` command, but it doesn't show the account's subcommands `new` and `update`.
```
$ mycommand account --help
Usage: mycommand account [OPTIONS]

Optional arguments:
  -h, --help  print help message
```

This behavior is identical to `--help account` which is my desired behavior, based on a common pattern with cli tools. Fixing the bug in the `--help account` approach should maintain this synonymous behavior for this case.
```
$ mycommand account --help new
Usage: mycommand account [OPTIONS]

Optional arguments:
  -h, --help  print help message
```

This shows the same output as without the `new`. Instead I request that it shows the full command as `mycommand account new [option]` plus the `new` specific options.
```
$ mycommand account new --help
Usage: mycommand account [OPTIONS]

Optional arguments:
  -h, --help  print help message
```

Just as with the `account --help` and `--help account` cases, this case should be synonymous with the previous case to `account --help new`.

Finally, this request should apply to an arbitrary depth of subcommands. Is this request coherent and possible? Are there any problems with this design or approach?

**Comment by murarth** [comment]
Thanks for the issue. You're absolutely correct that subcommand usage was broken.

I've just pushed a commit that fixes the behavior of `parse_args(_default)_or_exit` functions. It required some changes to the `Options` trait, including the addition of methods `self_usage` and `self_command_list` (which applications may also use to display usage properly).

I've also added a test similar to your example above, to ensure that the new functionality works properly with nested commands.

Let me know if you encounter any issues with the new functionality.
